### DISCUSSION TITLE
Use of multiple GPUs in XPU mode?

### DISCUSSION THREAD

**Discussion by samhodge** [dreamworksanimation/openmoonray#75] [Q&A]
I am lucky to have a machine with two RTX 3090 PCI cards with an NVLink bridge

When rendering in `-exec_mode xpu` I am only seeing 25% utilisation on one GPU, with around 4Gb of the possible 24Gb on that GPU in use.

Is it possible to get more GPU utilisation?

My test scene is the ALab2 scene, which without xpu mode takes about 1hr 40min to render.

**Comment by boberfly** [comment]
@samhodge what's the speed with XPU vs without, and what CPU do you use? Just curious, based on https://github.com/dreamworksanimation/openmoonray/discussions/50 I think the GPU will be throttled waiting on the CPU's need of shadow rays, and probably would only be increased in utilisation the more complex your scene is (to use all 24GB VRAM) and how many shadow-casting lights you have in the scene (to use the RT cores).

My assumption here is there's probably a balance of what your main system can handle in RAM in terms of scene complexity + textures + AOVs, and the GPU holding the OptiX BVH + scene geometry data that's needed for shadow rays in VRAM (position+motion steps, + maybe normals and opacity/alpha textures?) + and I guess also the AOV requirements for OptiX denoising, but this would probably only be the framebuffer beauty+normals+albedo+motionvectors(?) uploaded CPU memory to GPU and independent of the scene data.

Anyways best to wait for the Dreamworks guys to answer correctly, I'm just assuming here... :)

**Comment by samhodge** [comment]
With XPU render times are 1hr 32 minutes without I see 1hr 41 minutes 

only utility of GPU up to 25% computation and only 4Gb of VRAM and the scene sits at about 29Gb in use without much else running on the system: Terminal, Gnome System Monitor and Google Chrome on CentOS 7.9 with 128Gb total on the system.

**Comment by kjeffery** [comment]
You're correct: we only use one GPU at the moment. However, we anticipate some significant changes to our GPU support in the future (is that ambiguous enough for you?).

**Reply by jmahovsky-dwa** [reply]
Yes, the GPU may not be fully utilized if the CPU doesn't have enough shadow rays to keep it busy.

One of our goals is to offload more processing onto the GPU to increase utilization and speed up.

The low GPU memory usage is expected, because for the shadow ray processing only the scene geometry needs to be on the GPU, and not textures.

**Reply by samhodge** [reply]
Happy to see more cores screaming for mercy, GPU cores scream the loudest!
